### DISCUSSION TITLE
Meeting Notes - 15/07/23

### DISCUSSION THREAD

**Discussion by mindoodoo** [ratatui/ratatui#316] [Meeting Notes]
# Meeting Notes

## Releasing

- Before next release &rarr; remove `Canvas` from `prelude` &rarr; #317 
- @orhun will write **another** [blog post](https://blog.orhun.dev/ratatui-0-21-0/) &rarr; [update](https://blog.orhun.dev/ratatui-0-22-0/)
  - Will be promoted on reddit &rarr; [update](https://www.reddit.com/r/rust/comments/151z59f/ratatui_0220_is_released_community_fork_of_tuirs/)
- We should self nominate for the [this week in rust](https://this-week-in-rust.org/) **crate of the week**
  - [update](https://users.rust-lang.org/t/crate-of-the-week/2704/1217?u=orhun)
- Aim for *slightly* more frequent releases
  - In combination with nightly releasing

### Nightly releases

- General consensus is that we should set up nightly releases
- Frequency &rarr; once per week
  - With option to trigger manually in case by case basis

## Maintainers

- Need more &rarr; 2 ish
- Permissions should be reviewed before adding new maintainers
- A couple good candidates already manifested themselves
  - Will get back to individuals ASAP
- Always looking for candidates &rarr; #151

## `tui-rs`

- New attempt should be made to ask for **ownership transfer**
- Process should be something like :
  - Initital contact
  - Follow up with **actionable** list of things that will be done
    - Officially deprecate `tui-rs`
    - Link back to `ratatui`
    - General project direction

## Breaking Changes

- **General goal** &rarr; balance breaking things and adding features
  - Current pace has been quite good
- Breaking changes should be approached on a **case by case basis**
  - Since we're still trying to get some legacy `tui-rs` users it makes sense to keep the `tui-rs` to `ratatui` migration process simple for the time being
- The approach to renaming `Spans` to `Line` has been quite good
  - `Spans` was renamed to `Line`
  - A type alias of `Line` named `Spans` was introduced to make change non breaking
  - This alias was marked as **deprecated**
  - Will be removed in next release or the one after
- > Don't break everything in one go
- Giving notice to user should **always be favoured** before breaking changes
  - deprecation is great for this &rarr; compiler warnings

## [Ratatui book](https://github.com/tui-rs-revival/ratatui-book)

- Currently led by @Nickiel12
- **Idea :** implement section on different tui application development approaches
- Will potentially start PR / issue workflow to get more people involved and reduce workload

## Discord

- Add pingable roll for maintainers
- New contributors channel (is there a time target for completing reviews?)

## Removing doc tests

- @lthoerner will create a discussion &rarr; #324 

## Ratatui Domain (ratatui.rs)

- Cloudflare support (?)
- Landing page design (cc: @Nickiel)

## Scheduling meetings

- Will create a poll
- Third saturday of each months ?
- Needs comments

## Misc

- Small tweaks should be pushed on the maintainer branch
  - Takes little time
  - Request changes often saturate the project with **WIP PRs**
- Add language around deprecation policy &rarr; @kd
  - > (e.g. "between versions, expect deprecation warnings ... between every two versions, expect breaking changes ... read so and so changelog here").

---

@kdheepak has a cute puppy :

![cute_pup](https://github.com/tui-rs-revival/ratatui/assets/10436331/709604d1-384f-4f32-b5c8-f480f67c1135)

*Was too lazy to resize image*

**Comment by joshka** [comment]
> Before next release → remove Canvas from prelude

https://github.com/tui-rs-revival/ratatui/pull/317

**Reply by mindoodoo** [reply]
Updated in the notes !

**Comment by orhun** [comment]
> Releasing

https://github.com/tui-rs-revival/ratatui/releases/tag/v0.22.0

> @orhun will write another [blog post](https://blog.orhun.dev/ratatui-0-21-0/)

https://blog.orhun.dev/ratatui-0-22-0/

> Will be promoted on reddit

https://www.reddit.com/r/rust/comments/151z59f/ratatui_0220_is_released_community_fork_of_tuirs/

> We should self nominate for the [this week in rust](https://this-week-in-rust.org/) crate of the week

https://users.rust-lang.org/t/crate-of-the-week/2704/1217?u=orhun

**Reply by mindoodoo** [reply]
Updated in the notes !

**Comment by lthoerner** [comment]
I made the doctests discussion #324

**Reply by mindoodoo** [reply]
Updated in the notes !
